docs/knowledge_base/policy_examples.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Policy Examples
+category: 6255e67693d5200013b1fa41
+hidden: false
+---
+
+# Examples
+
+The policy transforms your threat model into a description of why the job is being blocked. There are multiple ways to define why a job should be blocked.
+
+## Blocking an issue
+
+The most common reason to block a job is because one of the dependencies has a known issue within one of Phylum's risk domains.
+
+The following policy shows ways to block issues based on a per-domain threshold.
+
+```rego
+package policy
+
+import data.phylum.domain
+import data.phylum.level
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+issue contains "risk level cannot exceed medium" if {
+	data.issue.domain in {domain.AUTHOR, domain.ENGINEERING, domain.VULNERABILITY}
+	data.issue.severity > level.MEDIUM
+}
+
+issue contains "malicious risk level cannot exceed low" if {
+	data.issue.domain == domain.MALICIOUS
+	data.issue.severity > level.LOW
+}
+
+issue contains "license risk level cannot exceed high" if {
+	data.issue.domain == domain.LICENSE
+	data.issue.severity > level.HIGH
+}
+```
+
+Given the following input:
+
+```json
+{
+    "dependencies": [{
+        "ecosystem": "pypi",
+        "id": "07a3feb6-d6fb-523a-88f0-9896b86e6f93",
+        "issues": [{
+            "domain": "malicious",
+            "severity": 4,
+            "tag": "CM0004"
+        }],
+        "issues_complete": true,
+        "name": "example-package",
+        "version": "1.0.0"
+    }]
+}
+```
+
+When the policy fails, the output will look something like this:
+
+```json
+{
+  "dependencies": [
+    {
+      "errors": [],
+      "id": "07a3feb6-d6fb-523a-88f0-9896b86e6f93",
+      "issues": [
+        {
+          "reason": "malicious risk level cannot exceed low",
+          "tag": "CM0004"
+        }
+      ]
+    }
+  ],
+  "errors": []
+}
+```
+
+When Phylum sees this output from the policy, it will block the job and generate a report naming the package and describing the issue.
+
+## Blocking a dependency
+
+If a dependency is being blocked for some other reason, the policy can indicate this by using the `errors` property on the dependency in the output.
```

Comment:
This is still technically correct but not useful anymore. `phylum.rego` sets the `errors` property of the dependency based on the `dependency` rule.

So in this example the policy sets `data.policy.dependency := ["AGPL licensed software is not allowed."` (for this particular dependency) and then `phylum.rego` picks that up and puts it into the relevant `data.phylum.policy.dependencies[_].errors`.

The `errors` property is still visible to users, but the way they block dependencies is by using `dependency` rules like this.

Reply:
Updated